### DISCUSSION TITLE
[tabufline] configure displayed text when two buffer has same names

### DISCUSSION THREAD

**Discussion by Reverier-Xu** [NvChad/NvChad#2907] [Q&A]
![image](https://github.com/NvChad/NvChad/assets/41937333/9ecf355d-426c-47d8-9856-766cb9c6d675)

Hi, I'm using NvChad for developing. I found that if I opened two files that have the same name, the tabs will display full relative paths on it, but the text is truncated, that makes me confused when I have many buffers opened. is that possible to config the display rule of tabufline? I'd like to have something like `<file_name>  <truncated relative path when buffers have same name>` on my tabs.

**Comment by siduck** [comment]
that will increase the tab size by a lot and currently all tab sizes are of 23 length, a lot of edgecases to handle :/

**Reply by Reverier-Xu** [reply]
Maybe we could make tab sizes configurable, and just truncate title strings that user constructed. 23 characters limitation doesn’t work well on all size of screens.

**Reply by siduck** [reply]
that wont help! 

you must be thinking it'd be easy to just show `filename dir1` `filename dir2`

what will happen when there are 3 files of same parent dir names?

~/.config/nvim/lua/abc/init.lua
~/config/nvim/lua/abc/abc/init.lua
~/.local/share/nvim/lua/abc/init.lua

**Reply by Reverier-Xu** [reply]
Yes you are right... we need a well designed display structure for that.

**Reply by siduck** [reply]
if we figure out running code on hovering any area of tabufline then we can show a lil floating window below the tab :thinking:  , with full name!
